src/components/Dialogs/Conditions/OrgUnitCondition.js

```diff
@@ -72,6 +79,111 @@ const OrgUnitCondition = ({
         />
     )
 }
+const getMetaDataId = (id) =>
+    ouIdHelper.hasLevelPrefix(id) || ouIdHelper.hasGroupPrefix(id)
+        ? ouIdHelper.removePrefix(id)
+        : id
+
+// VERSION-TOGGLE: rename to OrgUnitCondition when 42 is lowest supported version
+const OrgUnitConditionMinVersionV42 = ({
+    addMetadata,
+    addParentGraphMap,
+    condition,
+    onChange,
+    metadata,
+    parentGraphMap,
+}) => {
+    const { rootOrgUnits, currentUser } = useCachedDataQuery()
+    const selected = useMemo(() => {
+        const idsString = condition.split(':')?.[1] ?? ''
+
+        if (idsString.length === 0) {
+            return []
+        }
+
+        return idsString.split(';').map((id) => {
+            const isOuTreeItem =
+                !ouIdHelper.hasLevelPrefix(id) &&
+                !ouIdHelper.hasGroupPrefix(id) &&
+                !id.startsWith('USER_ORGUNIT')
+            const metaDataId = getMetaDataId(id)
+            const item = {
+                id,
+                // We store IDs for levels and groups in the metadata store without prefix
+                name:
+                    metadata[metaDataId].displayName ||
+                    metadata[metaDataId].name,
+            }
+
+            if (isOuTreeItem) {
+                item.path = getOuPath(id, metadata, parentGraphMap)
+            }
+
+            return item
+        })
+    }, [condition, metadata, parentGraphMap])
+    const onSelect = useCallback(
+        ({ items }) => {
+            if (items.length === 0) {
+                onChange('')
```

Comment:
What scenarios would this happen in?

Reply:
This is not hypothetical, it happens when the only one item is selected and it then gets deselected. An when this happens we do need to call `onChange` with an empty string, and not with the `IN` operator.